tests/cdi/cdi_test.go

```diff
@@ -0,0 +1,159 @@
+	. "github.com/onsi/gomega"
+)
+
+// template parameters
+const (
+	pvcEPHTTPNOAUTHURL  = "https://download.cirros-cloud.net/0.4.0/cirros-0.4.0-x86_64-disk.img"
+	pvcName        = "golden-pvc"
+	vmName         = "test-vm"
+	rawPVCFilePath = "tests/cdi/golden-pvc.yml"
+	rawVMFilePath  = "tests/cdi/test-vm.yml"
+	sepP = "-p"
+)
+
+var KubeVirtOcPath = ""
+
+var _ = Describe("CDI create importer-pod write images into pv", func() {
+	flag.Parse()
+
+	overrideTemplateFromParameters := func(srcFilePath, DstFilePath string, params ...string) string {
+		By(fmt.Sprintf("Overriding template from %s to %s", srcFilePath, DstFilePath))
+		args := []string{"process", "-f", srcFilePath}
+		args = append(args, params...)
+		out, err := RunOcCommand(args...)
+		Expect(err).ToNot(HaveOccurred())
+		filePath, err := WriteJson(DstFilePath, out)
+		Expect(err).ToNot(HaveOccurred())
+		return filePath
+	}
+
+	createResourceWithFilePath := func(resourceName, resourceType, filePath string) {
+		var message = ""
+		By(fmt.Sprintf("Creating %s:%s from JSON file via oc-create command", resourceType, resourceName))
+		out, err := RunOcCommand("create", "-f", filePath)
+		Expect(err).ToNot(HaveOccurred())
+		message = fmt.Sprintf(resourceType+" \"%s\" created\n", resourceName)
+		Expect(out).To(Equal(message))
+
+		Eventually(func() bool {
+			out, err := RunOcCommand("get", resourceType)
+			Expect(err).ToNot(HaveOccurred())
+			return strings.Contains(out, resourceName)
+		}, time.Duration(2)*time.Minute).Should(BeTrue(), "Timed out waiting for ovm to appear")
+	}
+
+	waitUntilResourceReady := func(resourceName, resourceType, query, expectOut string) {
+		By(fmt.Sprintf("Wait until %s:%s ready", resourceType, resourceName))
+		Eventually(func() bool {
+			//oc get resource with -o jsonpath query
+			out, err := RunOcCommand("get", resourceType, resourceName, query)
+			Expect(err).ToNot(HaveOccurred())
+			return strings.Contains(out, expectOut)
+		}, time.Duration(2)*time.Minute).Should(BeTrue(), "Timed out waiting for vm to appear")
+	}
+
+	Context("with oc command", func() {
+		dstPVCFilePath := ""
+		dstVMFilePath := ""
+
+		BeforeEach(func() {
+			SkipIfNoOc()
+			dstPVCFilePath = "/tmp/test-pvc"
+			dstVMFilePath = "/tmp/test-vm"
+		})
+
+		AfterEach(func() {
+			_, err := RunOcCommand("delete", "pvc", pvcName)
+			Expect(err).ToNot(HaveOccurred())
+
+			_, err = RunOcCommand("delete", "pod", "-l", "app=containerized-data-importer")
+			Expect(err).ToNot(HaveOccurred())
+
+			_, err = RunOcCommand("delete", "vm", vmName)
+			Expect(err).ToNot(HaveOccurred())
+
+			if dstPVCFilePath != "" {
+				// remove testing pvc template
+				err := os.Remove(dstPVCFilePath + ".json")
+				Expect(err).ToNot(HaveOccurred())
+				dstPVCFilePath = ""
+
+				// remove testing vm template
+				err = os.Remove(dstVMFilePath + ".json")
+				Expect(err).ToNot(HaveOccurred())
+				dstVMFilePath = ""
+			}
+		})
+
+		It("create pvc and vm should success", func() {
```

Comment:
I think that this assertion should be split into three parts (so in case of a failure we will know where to look):
- Verify that CDI populated the PVC
- Verify that the PVC is attached to the VM.
- Verify that the VM is functional (maybe using SSH?) - this way we ensure that the image didn't get corrupted during the import process.

Reply:
Okay, I already split it into four steps.
For the last one, we have some lib in kubevirt [here](https://github.com/kubevirt/kubevirt/blob/master/tests/vmi_networking_test.go#L204) I guess we could use, I will add it in a follow-up pr @gbenhaim wdyt ?
> Verify that the VM is functional (maybe using SSH?)